include/oboe/AudioStreamBuilder.h

```diff
@@ -179,6 +179,9 @@ class AudioStreamBuilder : public AudioStreamBase {
      *
      * By default, the primary device will be used.
      *
+     * Note that when using OpenSL ES, this will be ignored and the subsequent
```

Comment:
Is there an easy way for the client to find out whether Oboe will be using AAudio or OpenSL ES? I haven't found one.

Since the result of choosing a wrong device will be audible to end user, I suggest providing a better service here other than simply ignoring client's requests.

We can introduce "capabilities" that say which operations are possible and which are not. Though, that looks like a big change so can be addressed later. It's actually even better than simply telling the app if Oboe uses AAudio or OpenSL ES because there can be "partial" cases like we have with MMap mode.